docs/site/Loopback-component-authorization.md

```diff
@@ -8,543 +8,377 @@ permalink: /doc/en/lb4/Loopback-component-authorization.html
-
-```ts
-export const enum AuthorizeErrorKeys {
-  NotAllowedAccess = 'Not Allowed Access',
+  async authorize(
+    context: AuthorizationContext,
+    metadata: AuthorizationMetadata,
+  ) {
+    events.push(context.resource);
+    if (
+      context.resource === 'OrderController.prototype.cancelOrder' &&
+      context.principals[0].name === 'user-01'
+    ) {
+      return AuthorizationDecision.DENY;
+    }
+    return AuthorizationDecision.ALLOW;
+  }
 }
 ```
 
-Finally, we put everything together in one index file.
+The `AuthorizationContext` parameter of the `authorize` function contains the
+current principal (in the example given above,that would be the current user
+invoking `cancelOrder`) and details of the invoked endpoint. The
+`AuthorizationMetadata` parameter of the `authorize` function contains all the
+details provided in the invoked method's decorator.
 
-{% include code-caption.html content="/src/authorization/index.ts" %}
+> and then tagged to an application as `AuthorizationTags.AUTHORIZER` as below.
 
 ```ts
-export * from './component';
-export * from './types';
-export * from './keys';
-export * from './error-keys';
-export * from './permission-key';
-export * from './decorators/authorize.decorator';
-export * from './providers/authorization-metadata.provider';
-export * from './providers/authorization-action.provider';
-export * from './providers/user-permissions.provider';
+import AuthorizationTags from '@loopback/authorization';
+let app = new Application();
+app
+  .bind('authorizationProviders.my-provider')
+  .toProvider(MyAuthorizationProvider)
+  .tag(AuthorizationTags.AUTHORIZER);
 ```
 
-That is all for the authorization component. You can create all of the above
-into a loopback extension as well. Everything remains the same. Refer to the
-[extension generator](./Extension-generator.md) guide for creating an extension.
-
-## Usage
-
-In order to use the above component into our REST API application, we have a few
-more steps to go.
-
-- Add component to application.
-
-{% include code-caption.html content="/src/application.ts" %}
+- This creates a list of `authorize()` functions.
+  - The `authorize(AuthorizationContext, AuthorizationMetadata)` function in the
+    provider class is expected to be called by the `Authorization Interceptor`
+    which is called for every API endpoint decorated with `@authorize()`.
+  - The authorize interceptor gets the list of functions tagged with
+    `AuthorizationTags.AUTHORIZER` (and also the voters listed in the
+    `@authorize` decorator per endpoint) and calls the functions one after
+    another.
+  - The `authorize()` function is expected to return an object of type
+    `AuthorizationDecision`. If the type returned is
+    `AuthorizationDecision.ALLOW` the current `Principal` has passed the
+    executed `authorize()` function's criteria.
+
+> voter functions are directly provided in the decorator of the remote method
```

Comment:
what sentence does `voter functions are directly` belong to? And why does it start with '>'  which idents it?

![image](https://user-images.githubusercontent.com/6864736/70177227-fab90380-16a7-11ea-852e-63b4c07a220f.png)

Reply:
I thought highlighting this way , gives better readability